app/src/main/java/at/bitfire/icsdroid/ProcessEventsTask.kt

```diff
@@ -77,7 +78,9 @@ class ProcessEventsTask(
             Log.d(Constants.TAG, "Removing all alarms from ${uid}: $this")
             alarms.clear()
         }
-        subscription.defaultAlarmMinutes?.let { minutes ->
+        val allDay = net.fortuna.ical4j.model.property.Duration(Duration.ofHours(24))
+        val isAllDay = event.duration == allDay
+        (if (isAllDay) subscription.defaultAllDayAlarmMinutes else subscription.defaultAlarmMinutes)?.let { minutes ->
```

Comment:
Use another value declaration for clarity. `let {}` is mostly useful when it's used to assign the result to a value (`val a = b.let {} …`). If there's no assigment, we can use `if` for better readability.

```
val isAllDay = DateUtils.isDate(dtStart)
val alarmMinutes = if (isAllDay)
    subscription.defaultAllDayAlarmMinutes
else
    subscription.defaultAlarmMinutes
if (alarmMinutes != null) { minutes -> ...
```